tools/depcheck/pkg/cmd/trace.go

```diff
@@ -0,0 +1,251 @@
+			parsedRoots = append(parsedRoots, entry)
+			continue
+		}
+
+		slash := ""
+		if string(entry[len(entry)-1]) != "/" {
+			slash = "/"
+		}
+		entry = strings.Join([]string{entry, slash, "..."}, "")
+		parsedRoots = append(parsedRoots, entry)
+	}
+
+	return parsedRoots
+}
+
+func (o *TraceOptions) Validate() error {
+	if len(o.GraphOptions.Entrypoints) == 0 {
+		return errors.New("at least one root package must be provided")
+	}
+	if len(o.OutputFormat) > 0 && o.OutputFormat != "dot" {
+		return fmt.Errorf("invalid output format provided: %s", o.OutputFormat)
+	}
+
+	return nil
+}
+
+// Run execs `go list` on all package entrypoints specified through --root.
+// Each package's ImportPath and non-transitive (immediate) imports are
+// filtered and aggregated. A package is filtered based on whether its ImportPath
+// matches an accepted pattern defined in the set of validPackages.
+// Each aggregated package becomes a node in a generated dependency graph.
+// An edge is created between a package and each of its Imports.
+func (o *TraceOptions) Run() error {
+	pkgs, err := depgraph.CalculatePackages(
+		ensureEntrypointEllipsis(
+			ensureVendorEntrypoint(o.GraphOptions.Entrypoints, o.GraphOptions.RepoRoot),
+		),
+	)
+	if err != nil {
+		return err
+	}
+
+	g, err := depgraph.BuildGraph(pkgs, o.GraphOptions.Entrypoints, o.GraphOptions.Excludes)
+	if err != nil {
+		return err
+	}
+
+	if len(o.GraphOptions.Filters) > 0 {
+		g, err = depgraph.FilterPackages(g, o.GraphOptions.Filters)
+		if err != nil {
+			return err
+		}
+	}
+
+	// possibleRoots are nodeNames that should be excluded
+	// from pruning or from analysis. We cannot rely on our
+	// given entrypoints as our source of roots because
+	// nodes created for package entrypoints are likely
+	// to have incomplete edge data based on the .go files
+	// immediately present in an entrypoint directory, and
+	// the direct import paths that they define.
+	possibleRoots := []string{}
+	for _, n := range g.Nodes() {
+		node, ok := n.(*depgraph.Node)
+		if !ok {
+			continue
+		}
+		if !isOriginPackage(node, o.GraphOptions.RepoRoot) {
+			continue
+		}
+		if len(g.To(n)) > 0 || len(g.From(n)) == 0 {
+			continue
+		}
+
+		possibleRoots = append(possibleRoots, node.UniqueName)
+	}
+
+	// prune unreachable nodes from graph
+	g.PruneOrphans(possibleRoots)
+	return o.outputGraph(g)
+
+}
+
+func (o *TraceOptions) outputGraph(g graph.Directed) error {
+	if o.OutputFormat != "dot" {
+		return fmt.Errorf("invalid output format provided: %s", o.OutputFormat)
+	}
+
+	data, err := dot.Marshal(g, fmt.Sprintf("%q", strings.Join(o.GraphOptions.Entrypoints, ", ")), "", " ", false)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(o.Out, "%v\n", string(data))
+	return nil
+}
+
+// isOriginPackage receives a *graph.Node and the repo root import path
+// Returns true if the given node's label has the repo root as its prefix.
+func isOriginPackage(n *depgraph.Node, rootName string) bool {
```

Comment:
that's not what this method is.  Doesn't look like origin.  Looks like `isNotVendor`